transcom/mymove
# B 18317 multi move get all moves endpoint

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/Default.aspx?menu=StoryBoardPage)

## Summary

This story introduces a new endpoint on the internal API to retrieve all moves for a service member. The moves are returned in a json object with structure {currentMove: [{}], previousMoves: [{}]}

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Create a service member and go through the process of requesting a move.
2. Approve the move as a counselor and approve the move as a TOO
3. Add shipments to the move as a prime user
4. Through Postman or swagger send a request to the /all_moves{serviceMemberId} endpoint of the internalAPI
5. You should get a response with an JSON object {currentMove: [{}], previousMoves: [{}]}. currentMove will have one move in it representing the move with the latest CreatedAt DateTime. PreviousMoves will have all other moves.
6. To test previous moves make sure the service member has more than one move


### Backend



## Screenshots

![image](https://github.com/transcom/mymove/assets/136514590/f99d53d3-1891-499f-bef7-e4a474e471e3)![image](https://github.com/transcom/mymove/assets/136514590/6f185b20-3b60-462f-9e5c-4c18c20ea054)
